apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerException.cs

```diff
@@ -101,7 +119,20 @@ internal SpannerException(ErrorCode code, RpcException innerException)
             : base(GetMessageFromErrorCode(code), innerException)
         {
             Logger.DefaultLogger.LogPerformanceCounter("SpannerException.Count", x => x + 1);
+
+            SessionExpired = innerException.IsSessionExpiredError();
             ErrorCode = innerException.IsSessionExpiredError() ? ErrorCode.Aborted : code;
+
+            Metadata.Entry retryInfoEntry = innerException.Trailers.FirstOrDefault(
+                entry => s_retryInfoMetadataKey.Equals(entry.Key, StringComparison.InvariantCultureIgnoreCase));
+            if (retryInfoEntry != null)
+            {
+                RetryInfo retryInfo = RetryInfo.Parser.ParseFrom(retryInfoEntry.ValueBytes);
+                if (retryInfo?.RetryDelay?.Nanos >= 0)
```

Comment:
I suspect this was copied from some of my code somewhere, but I think it's slightly broken - if the server specifies a whole number of seconds, this won't pick that up.
I'd convert it to a TimeSpan unconditionally, and only set the RecommendedRetryDelay if that's not TimeSpan.Zero.

Reply:
Done. I'll look the code that was doing this then, because it probably needs to be fixed as well.